In [2]:
import pandas as pd
import numpy as np
from ast import literal_eval

In [3]:
df = pd.read_csv('wondoofin.csv')
cat_df = pd.read_csv('category.csv')

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 559 entries, 0 to 558
Data columns (total 26 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 559 non-null    int64 
 1   이름                 559 non-null    object
 2   로스터리               559 non-null    object
 3   타입                 559 non-null    object
 4   로스팅 포인트            559 non-null    object
 5   지속가능성              113 non-null    object
 6   컵 노트               559 non-null    object
 7   바디감                559 non-null    int64 
 8   신맛                 559 non-null    int64 
 9   단맛                 559 non-null    int64 
 10  쓴맛                 559 non-null    int64 
 11  커피 소개              559 non-null    object
 12  나라                 543 non-null    object
 13  로스터리 소개            559 non-null    object
 14  로스터리 주소            550 non-null    object
 15  이름.1               559 non-null    object
 16  식품의 유형             554 non-null    object
 1

In [7]:
df.columns

Index(['id', '이름', '로스터리', '타입', '로스팅 포인트', '지속가능성', '컵 노트', '바디감', '신맛', '단맛',
       '쓴맛', '커피 소개', '나라', '로스터리 소개', '로스터리 주소', '이름.1', '식품의 유형',
       '제조원 및 소재지', '유통기한', '제조일자', '내용량', '보관 방법', '원재료 및 함량',
       '제품문의 관련 주소 및 전화번호', '카페인', '로스터리ID'],
      dtype='object')

In [6]:
df.shape

(559, 26)

***
null 값 처리
- 지속가능성 : 113
- 나라 : 543
***
- 로스터리 주소 : 550
- 제조원 및 소재지 : 557/559
- 유통기한 : 557
- 보관 방법 : 545
- 식품의 유형 : 554
***

In [5]:
# 리스트 처리
df['컵 노트'] = df['컵 노트'].apply(literal_eval)
df['컵 노트'][0]

['토피', '맥아', '흑설탕']

***

***
추천과는 상관없는 칼럼들
1. 주소
***

In [9]:
print(df.loc[df['로스터리 주소'].isna(), 'id'].tolist())

[4701, 4692, 4613, 4614, 4695, 4694, 4612, 4611, 4693]


In [10]:
df.loc[df['로스터리 주소'].isna(), '로스터리 주소'] = 0

In [11]:
print(df.loc[df['제조원 및 소재지'].isna(), 'id'].tolist())

[274, 272]


In [12]:
print('로스터리 : ', df.loc[df['id'] == 4701, '로스터리'])
print('로스토리 주소 : ', df.loc[df['id'] == 4701, '로스터리 주소'])
print('제조원 : ', df.loc[df['id'] == 4701, '제조원 및 소재지'])

로스터리 :  17    커피네츄럴스페셜티
Name: 로스터리, dtype: object
로스토리 주소 :  17    0
Name: 로스터리 주소, dtype: object
제조원 :  17    경기도 고양시 덕양구 행신동 741-17 1층
Name: 제조원 및 소재지, dtype: object


***
df의 로스터리 null 정보 확인
***

In [53]:
addrdf = df[['로스터리ID', '로스터리', '로스터리 주소', '제조원 및 소재지']]
addrdf.head()

로스터리ID         로스터리                                         로스터리 주소  \
0      57        언더프레셔                           서울 강남구 영동대로106길 37 1층   
1      67         운조커피                          경기도 화성시 동탄지성로 306번길 68   
2      77    제이엠디커피팩토리  경기 고양시 일산동구 장대길 75-53 나동 1층 JMD Coffee Factory   
3      11  뉴웨이브 커피로스터스                           서울특별시 양천구 오목로56길 2,1층   
4      34     베르크 로스터스                          부산 부산진구 전포동 356-6 지하1층   

                                        제조원 및 소재지  
0                    (주)핸디엄, 경기도 여주시 세종대왕면 상동길 68  
1                          경기도 화성시 동탄지성로 306번길 68  
2  경기 고양시 일산동구 장대길 75-53 나동 1층 JMD Coffee Factory  
3        뉴웨이브커피로스터스, 서울시 양천구 오목로 56길 2, 영원빌딩 103호  
4             베르크로스터스, 부산광역시 사상구 학감대로222번길 83, 2층

In [54]:
result = addrdf[addrdf['로스터리 주소']==0]
result

Empty DataFrame
Columns: [로스터리ID, 로스터리, 로스터리 주소, 제조원 및 소재지]
Index: []

In [56]:
# result2 = addrdf[addrdf['제조원 및 소재지'].isna()]
result2 = addrdf[addrdf['제조원 및 소재지']==0]
result2

로스터리ID   로스터리                 로스터리 주소 제조원 및 소재지
413      42  센터 커피  서울특별시 성동구 서울숲 2길 28-11         0
529      42  센터 커피  서울특별시 성동구 서울숲 2길 28-11         0

In [22]:
result3 = addrdf[addrdf['제조원 및 소재지'] != addrdf['로스터리 주소']]
result3

로스터리ID         로스터리                    로스터리 주소  \
0        57        언더프레셔      서울 강남구 영동대로106길 37 1층   
3        11  뉴웨이브 커피로스터스      서울특별시 양천구 오목로56길 2,1층   
4        34     베르크 로스터스     부산 부산진구 전포동 356-6 지하1층   
6        18    라메르판지 비앤알              울산 북구 판지1길 30   
7        98         토치커피        서울시 강남구 봉은사로 68길 41   
..      ...          ...                        ...   
541      93       컨트리사이드  경기도 고양시 덕양구 행당로 1 4층 401호   
543      48   식물학 커피로스터스        서울시 강남구 강남대로154길 33   
544      49      싸인포스트커피           서울시 동작구 솔밭로 97-6   
553      61         오드이븐    부산 중구 중구로29번길 7 1층 오드이븐   
558      36  보사노바 커피로스터스          강원 강릉시 창해로14번길 28   

                                     제조원 및 소재지  
0                 (주)핸디엄, 경기도 여주시 세종대왕면 상동길 68  
3     뉴웨이브커피로스터스, 서울시 양천구 오목로 56길 2, 영원빌딩 103호  
4          베르크로스터스, 부산광역시 사상구 학감대로222번길 83, 2층  
6    라메르판지 비앤알 / 울산광역시 북구 판지1길 30 (지상2~3층 구유동)  
7                      서울시 강남구 봉은사로 68길 41, 1층  
..                                         ...  
541                   경기도 고양시 덕양구 행당로1 4층 401호  
543         식물학커피로스터스, 서울특별시 용산구 서빙고로67길 21 2층  
544                  싸인포스트커피, 서울시 동작구 솔밭로 97-6  
553                   부산광역시 중구 중구로 29번길 7 오드이븐  
558            보사노바로스팅팩토리, 강원도 강릉시 경강로2660번길 7  

[318 rows x 4 columns]

<로스터리 주소와 제조원이 항상 일치하지는 않는다.> <br>
로스터리 주소 : 커피네츄럴스페셜티, 헤베커피 <br>
제조원은 그냥 0으로 둬야할 것 같다.

In [46]:
df.loc[df['로스터리'] == '커피네츄럴스페셜티', '로스터리 주소'] = df['제조원 및 소재지'][df['로스터리'] == '커피네츄럴스페셜티'].values[0]
df.loc[df['로스터리'] == '헤베커피', '로스터리 주소'] = df['제조원 및 소재지'][df['로스터리'] == '헤베커피'].values[0]

In [35]:
df['제조원 및 소재지'][df['로스터리'] == '커피네츄럴스페셜티'].values[0]

'경기도 고양시 덕양구 행신동 741-17 1층'

In [52]:
df.loc[df['제조원 및 소재지'].isna(), '제조원 및 소재지'] = 0

***
위 확인 블락을 다시 돌려 주소null값이 없고 제조원 null값이 0으로 바뀌었는지 확인!
***

***
식품의 유형, 유통기한, 보관 방법
***